### PR TITLE
Fixed PUYA flash write buffer alignment

### PR DESCRIPTION
**Pull request by drzony**
This merge fixes flash writing on PUYA flashes when write size is not a multiple of 4.
This bug prevents updating via Arduino OTA when using gzipped image.

### CODE REVIEW

File: cores/esp8266/Esp.cpp [hunk 1]
```diff
@@ -697,15 +697,16 @@ static SpiFlashOpResult spi_flash_write_puya(uint32_t offset, uint32_t *data, si
         } else {
             bytesLeft = 0;
         }
-        rc = spi_flash_read(pos, flash_write_puya_buf, bytesNow);
+        size_t bytesAligned = (bytesNow & 3) ? ((bytesNow & ~3) + 4) : bytesNow;
```
Comment (earlephilhower) [inline cores/esp8266/Esp.cpp:700]: I think it's more commonly done as
````
         size_t bytesAligned = (bytesNow + 3)  & ~3; 
````

Comment (drzony) [reply cores/esp8266/Esp.cpp]: Yup, got a little brain freeze there 😄

**Comment by yumkam** [comment]
FTR, this over-reads `data` by up to 3 bytes. Assuming `data` is 4-bytes aligned and does not point to some kind of "special" memory (mmio/etc), this should be practically harmless. But formally this is still undefined behavior and compiler is free to kill kittens and start WWIII.

**Comment by drzony** [comment]
Thanks for the observation @yumkam , must have missed this somehow. Another pull request is on the way.

**Comment by drzony** [comment]
As for the overreading I have not seen a flash that is not 4-byte aligned. All the functions for spi_flash ops have `uint32 *` as parameters.

**Pull request merged**